### ISSUE TITLE
Generate/Consume 'table' values from CSV Reader/Writer

### ISSUE THREAD

**Issue by lafernando** [ballerina-platform/ballerina-standard-library#408]
$subject. Currently CSV works with just byte[], it would be better if we have enhanced functionality where it works with 'table' types, and allows it to be conveniently used with queries.

**Comment by daneshk** [comment]
New high-level CSV APIs provide support to read CSV content as streams[1]. Please check whether satisfies the requirement. 

1. https://dev.ballerina.io/learn/by-example/csv-io.html

**Comment by lafernando** [comment]
> New high-level CSV APIs provide support to read CSV content as streams[1]. Please check whether satisfies the requirement.
> 
> 1. https://dev.ballerina.io/learn/by-example/csv-io.html

Not really, the main requirement is to get a data structure that has columns named so we can query them easily, e.g. in language -integrated queries. So still a string[] stream does not solve this, it's still similar to the earlier available functionality. So the most natural data structure I see for this requirement is to use a table.

**Comment by daneshk** [comment]
We are now reading the CSV value as a record[] now. Hence, closing this issue

**Comment by lafernando** [comment]
Great!
